src/core/spark/src/main/scala/TransformParam.scala

```diff
@@ -53,4 +53,12 @@ class TransformerArrayParam(parent: String, name: String, doc: String, isValid:
   /** Creates a param pair with the given value (for Java). */
   def w(value: java.util.List[Transformer]): ParamPair[Array[Transformer]] = w(value.asScala.toArray)
 
+  override def jsonEncode(value: Array[Transformer]): String = {
+    throw new NotImplementedError("The transform cannot be encoded.")
```

Comment:
This is the same pattern our other complex parameters have, unfortunately there is no straightforward way to json encode an arbitrary transfromer